sdk/compiler/damlc/daml-stdlib-src/DA/Bytes.daml

```diff
@@ -0,0 +1,71 @@
+-- Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF 2.dev
+
+{-# LANGUAGE CPP #-}
+-- | Data types and functions for working with hex encoded strings.
+module DA.Bytes
+  ( BytesHex
+  , bytesHex
+  , Bytes32Hex
+  , bytes32Hex
+  , UInt32Hex
+  , uint32Hex
+  , UInt64Hex
+  , uint64Hex
+  , UInt256Hex
+  , uint256Hex
+  ) where
+
+import DA.Text qualified as Text
+
+newtype BytesHex = BytesHex { getText: Text }
+  deriving Eq
+
+instance Show BytesHex where
+  show (BytesHex bytes) = bytes
+
+bytesHex : Text -> BytesHex
+bytesHex bytes | (Text.length bytes) % 2 == 0 && Text.isHex bytes = BytesHex bytes
+bytesHex bytes = error $ "BytesHex should only contain hex encoded bytes as members, given " <> bytes
+
+newtype Bytes32Hex = Bytes32Hex { getBytesHex: BytesHex }
+  deriving Eq
+
+instance Show Bytes32Hex where
+  show (Bytes32Hex bytes) = show bytes
+
+bytes32Hex : Text -> Bytes32Hex
+bytes32Hex bytes | Text.length bytes == 2*32 = Bytes32Hex (bytesHex bytes)
+bytes32Hex bytes = error $ "Bytes32Hex should only have 32 bytes, given " <> show bytes
```

Comment:
At the moment, we only have requirements for:
- equality comparison
- string representation of the data structure
- byte concatenation

By using `newtype` to introduce these types, is it reasonable to assume that we can change the underlying concrete data representation at a future point in time (e.g. to use Bytes, ByteString, etc)?